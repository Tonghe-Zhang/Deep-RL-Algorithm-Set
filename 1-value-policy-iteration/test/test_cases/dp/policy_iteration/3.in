{
    "P": {
        "0": {
            "0": [
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ]
            ]
        },
        "1": {
            "0": [
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ]
            ]
        },
        "2": {
            "0": [
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    1,
                    0.0,
                    false
                ]
            ]
        },
        "3": {
            "0": [
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    3,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ]
            ]
        },
        "4": {
            "0": [
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    0,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ]
            ]
        },
        "5": {
            "0": [
                [
                    1.0,
                    5,
                    0,
                    true
                ]
            ],
            "1": [
                [
                    1.0,
                    5,
                    0,
                    true
                ]
            ],
            "2": [
                [
                    1.0,
                    5,
                    0,
                    true
                ]
            ],
            "3": [
                [
                    1.0,
                    5,
                    0,
                    true
                ]
            ]
        },
        "6": {
            "0": [
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    7,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    2,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ]
            ]
        },
        "7": {
            "0": [
                [
                    1.0,
                    7,
                    0,
                    true
                ]
            ],
            "1": [
                [
                    1.0,
                    7,
                    0,
                    true
                ]
            ],
            "2": [
                [
                    1.0,
                    7,
                    0,
                    true
                ]
            ],
            "3": [
                [
                    1.0,
                    7,
                    0,
                    true
                ]
            ]
        },
        "8": {
            "0": [
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    4,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ]
            ]
        },
        "9": {
            "0": [
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    5,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    8,
                    0.0,
                    false
                ]
            ]
        },
        "10": {
            "0": [
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    11,
                    0.0,
                    true
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    11,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    11,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    6,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ]
            ]
        },
        "11": {
            "0": [
                [
                    1.0,
                    11,
                    0,
                    true
                ]
            ],
            "1": [
                [
                    1.0,
                    11,
                    0,
                    true
                ]
            ],
            "2": [
                [
                    1.0,
                    11,
                    0,
                    true
                ]
            ],
            "3": [
                [
                    1.0,
                    11,
                    0,
                    true
                ]
            ]
        },
        "12": {
            "0": [
                [
                    1.0,
                    12,
                    0,
                    true
                ]
            ],
            "1": [
                [
                    1.0,
                    12,
                    0,
                    true
                ]
            ],
            "2": [
                [
                    1.0,
                    12,
                    0,
                    true
                ]
            ],
            "3": [
                [
                    1.0,
                    12,
                    0,
                    true
                ]
            ]
        },
        "13": {
            "0": [
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    9,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    12,
                    0.0,
                    true
                ]
            ]
        },
        "14": {
            "0": [
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ]
            ],
            "1": [
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    15,
                    1.0,
                    true
                ]
            ],
            "2": [
                [
                    0.3333333333333333,
                    14,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    15,
                    1.0,
                    true
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ]
            ],
            "3": [
                [
                    0.3333333333333333,
                    15,
                    1.0,
                    true
                ],
                [
                    0.3333333333333333,
                    10,
                    0.0,
                    false
                ],
                [
                    0.3333333333333333,
                    13,
                    0.0,
                    false
                ]
            ]
        },
        "15": {
            "0": [
                [
                    1.0,
                    15,
                    0,
                    true
                ]
            ],
            "1": [
                [
                    1.0,
                    15,
                    0,
                    true
                ]
            ],
            "2": [
                [
                    1.0,
                    15,
                    0,
                    true
                ]
            ],
            "3": [
                [
                    1.0,
                    15,
                    0,
                    true
                ]
            ]
        }
    },
    "nS": 16,
    "nA": 4,
    "gamma": 0.9,
    "eps": 0.001
}