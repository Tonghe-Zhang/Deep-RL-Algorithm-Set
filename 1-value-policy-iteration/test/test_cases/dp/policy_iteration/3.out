{
    "value_function": [
        0.06144364796286837,
        0.05515526474996747,
        0.06984367551148236,
        0.05087275421575311,
        0.08503352411276896,
        0.0,
        0.10970644744558003,
        0.0,
        0.1398383461464696,
        0.24364731933220632,
        0.29690306676942646,
        0.0,
        0.0,
        0.3770962341389107,
        0.6375103664762097,
        0.0
    ],
    "policy": [
        0,
        3,
        0,
        3,
        0,
        0,
        0,
        0,
        3,
        1,
        0,
        0,
        0,
        2,
        1,
        0
    ]
}